conductor/src/lib.rs

```diff
@@ -35,18 +35,30 @@ pub async fn generate_spec(
     namespace: &str,
     backups_bucket: &str,
     spec: &CoreDBSpec,
-) -> Value {
+) -> Result<Value, ConductorError> {
     let mut spec = spec.clone();
+    let cloud_provider = data_plane_id.split('_').next().unwrap_or("");
```

Comment:
I think parsing the cloud provider from the data plane ID is not the best way to know which cloud provider is being used

Reply:
I am open to suggestions to how to work around it.  Currently the only data plane related info we get in Conductor is set as the `data_plane_id`.  If that isn't the best way maybe we could add a cloud provider string in the message from the control plane?  

We could do a look up the dataplane from the database, but we still couldn't use the `data_plane_id` to look that up since it's a string from the fields in the table.  Maybe just passing the provider id from the control-plane in the message is the right idea?

Reply:
Thinking about this from a self hosted perspective as well. In those cases we would need to add something that would allow for choosing only one cloud provider based on the environment it's been installed in.

Reply:
In that case we already set the environment variable of `is_cloudformation` or `is_gcp` in Conductor.  We will need something like that for Azure as well.  So I think we can just assume the cloud provider from those settings.

If both of you are fine with that I can get that added.

Reply:
I like that idea!

Reply:
The last commit addresses this and set the cloud provider based off the env var that is set.